cirq/circuits/circuit.py

```diff
@@ -171,7 +171,7 @@ def __getitem__(self, key: int) -> ops.Moment:
 
     def __getitem__(self, key):
         if isinstance(key, slice):
-            return Circuit(self._moments[key])
+            return Circuit(self._moments[key], self.device)
```

Comment:
should maybe add test to check that device is propagated

Reply:
Done.